src/Cli/dotnet/completions.ps1

```diff
@@ -0,0 +1,1383 @@
+using namespace System.Management.Automation
+using namespace System.Management.Automation.Language
+
+Register-ArgumentCompleter -Native -CommandName 'dotnet' -ScriptBlock {
+    param($wordToComplete, $commandAst, $cursorPosition)
+
+    $commandElements = $commandAst.CommandElements
+    $command = @(
+        'dotnet'
+        for ($i = 1; $i -lt $commandElements.Count; $i++) {
+            $element = $commandElements[$i]
+            if ($element -isnot [StringConstantExpressionAst] -or
+                $element.StringConstantType -ne [StringConstantType]::BareWord -or
+                $element.Value.StartsWith('-') -or
+                $element.Value -eq $wordToComplete) {
+                break
+            }
+            $element.Value
+        }) -join ';'
+
+    $completions = @()
+    switch ($command) {
+        'dotnet' {
+            $staticCompletions = @(
+                [CompletionResult]::new('--help', '--help', [CompletionResultType]::ParameterName, "Show command line help.")
```

Comment:
Are these auto-generated somehow? How are they maintained? Should the descriptions be localized?

Reply:
I actually need to remove this, it shouldn't be checked in here. I do need to add snapshots for the .NET CLI grammar to prevent unexpected drift now that you mention it!